lib/smart_on_fhir/utils.go

```diff
@@ -0,0 +1,69 @@
+package smart_on_fhir
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+type ehrAuthMetadata struct {
+	AuthEndpoint  string `json:"authorization_endpoint"`
+	TokenEndpoint string `json:"token_endpoint"`
+}
+
+func discoverAuthServer(issuer string, target interface{}) error {
+	resp, err := http.Get(issuer + "/.well-known/smart-configuration")
+
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	b, err := io.ReadAll(resp.Body)
+
+	return json.Unmarshal(b, target)
+}
+
+func RequestAccessToken(tokenUrl string, clientId string, clientSecret string, code string) (*FhirTokenResponse, error) {
+	client := &http.Client{
+		Timeout: time.Second * 10,
+	}
+
+	// Set the form data for the request
+	data := url.Values{}
+	data.Set("grant_type", "authorization_code")
+	data.Set("code", code)
+	data.Set("redirect_uri", "http://localhost:21001/fhir/epic/return/provider")
```

Comment:
We need to add a warning here that it's scoping directly to EPIC's return URL.

If we decide to use another EHR, then we need to dynamically generate the URL;

Reply:
We could add a providerName param and parse it like on `server/handlers/get_fhir_ehr_launch.go:15`